File: code/src/main/java/workout_manager/model/ExerciseAlt.java
package workout_manager.model;

/**
 * defines the exercise alt class for deserializing JSONS from the third part
 * API
 * 
 * @author group4
 */
public class ExerciseAlt {

    private String name;
    private String description;
    private int category;
    private int[] equipment;

    /**
     * creates an exerciseAlt object
     * 
     * @precondition !name.isEmpty() && description.isEmpty() && name != null &&
     *               description != null
     * @postcondition this.name == name && this.description == description
     * @param name        the name of the excersise
     * @param description the description of the exercise
     * @param category    the muscle category
     * @param equipment   int representation of equipment
     */
    public ExerciseAlt(String name, String description, int category, int[] equipment) {
        if (name == null) {
            throw new IllegalArgumentException("name cant be null created");
        }
        if (description == null) {
            throw new IllegalArgumentException("description cant be null created");
        }
        if (name.isEmpty()) {
            throw new IllegalArgumentException("exercise name cant be empty");
        }
        if (description.isEmpty()) {
            throw new IllegalArgumentException("exercise description cant be empty");
        }
        this.name = name;
        this.description = description;
        this.category = category;
        this.equipment = equipment;

    }

    /**
     * gets the name of the exercise
     * 
     * @return the name
     */
    public String getName() {
        return this.name;
    }

    /**
     * gets the text description of the exercise
     * 
     * @return the description of the exercise
     */
    public String getDescription() {
        return this.description.replaceAll("\\<[^>]*>", "");
    }

    /**
     * returns the int representation of the muscle category defined by the data
     * 
     * @return the int representation of the muscle category
     */
    public int getCategory() {
        return this.category;
    }

    /**
     * returns the int representation of the muscle category defined by the data
     * 
     * @return the int representation for equipment
     */
    public int getEquipment() {
        if (this.equipment.length > 0) {
            return this.equipment[0];
        } else {
            return 0;
        }

    }

}

File: code/Server/ExerciseAlt.py
#!/usr/bin/env python3

# *
# * defines the exercise alt class for deserializing JSONS
# * 
# * 
# * @author group4
class ExerciseAlt :
    
    # *
    #     * creates an exerciseAlt object
    #     * 
    #     * @precondition !name.isEmpty() && description.isEmpty() && name != null &&
    #     *               description != null
    #     * @postcondition this.name == name && this.description == description
    #     * @param name        the name of the excersise
    #     * @param description the description of the exercise
    #     * @param category    the muscle category
    #     * @param equipment   int representation of equipment
    #     * @param intensity   int representation of intensity
    def defineDetails(name,  description,  category,  equipment, intensity) :
        exerciseDetails = {}
        if (name == None) :
            raise  ("name cant be null created")
        if (description == None) :
            raise  ("description cant be null created")
        if not name :
            raise  ("exercise name cant be empty")
        if not description :
            raise  ("exercise description cant be empty")

        exerciseDetails["name"] = name
        exerciseDetails["description"] = description
        exerciseDetails["category"] = category
        exerciseDetails["equipment"] = equipment
        exerciseDetails["intensity"] = intensity
        return exerciseDetails

